File: certification_server/src/iface/common/registry_manager.go
package common

import commonmodel "certification_server/src/models/common"

// IRegistryManager 定义公共服务发现和注册相关操作。
type IRegistryManager interface {
	// Register 注册服务实例，ttl参数指定实例的生存时间，单位为秒。
	Register(instance *commonmodel.ServiceInstance, ttl int64) error
	// UnRegister 注销服务实例。
	UnRegister(instance *commonmodel.ServiceInstance) error
	// GetServiceInstances 获取指定服务名称的所有可用实例列表。
	GetServiceInstances(serviceName string) ([]*commonmodel.ServiceInstance, error)
	// GetServiceSnapShot 获取指定服务名称的当前服务快照信息，包含实例列表和版本等元数据。
	GetServiceSnapShot(serviceName string) (*commonmodel.ServiceSnapshot, error)
	// ChooseEndpoint 根据服务名称、亲和性键和要求的标签选择一个合适的服务实例，支持负载均衡和标签过滤等功能。
	ChooseEndpoint(serviceName string, affinityKey string, requireTags []string) (*commonmodel.ServiceInstance, error)
}

File: data_server/src/iface/common/registry_manager.py
from abc import ABC, abstractmethod

from src.models.common.instance import ServiceInstance, ServiceSnapshot


class IRegistryManager(ABC):
    """服务注册与发现管理接口。"""

    @abstractmethod
    async def register(self, instance: ServiceInstance, ttl_sec: int) -> None:
        """注册服务实例并维持心跳，ttl_sec 是注册过期时间，服务实例需要在过期前续约以保持可用。"""
        raise NotImplementedError

    @abstractmethod
    async def unregister(self, instance: ServiceInstance) -> None:
        """注销服务实例。"""
        raise NotImplementedError

    @abstractmethod
    async def get_service_instances(self, service_name: str) -> list[ServiceInstance]:
        """根据服务名称获取可用实例。"""
        raise NotImplementedError

    @abstractmethod
    async def get_service_snapshot(self, service_name: str) -> ServiceSnapshot | None:
        """根据服务名称获取当前快照。"""
        raise NotImplementedError

    @abstractmethod
    async def choose_endpoint(
        self,
        service_name: str,
        affinity_key: str = "",
        require_tags: list[str] | None = None,
    ) -> ServiceInstance | None:
        """根据服务名称、亲和性键和标签要求选择端点。"""
        raise NotImplementedError

